### PR TITLE
Revise "Upload your NFT file" screen

### PR DESCRIPTION
**Pull request by kiroj305**
- Remove "Choose your NFT format" screen
- Revise "Upload your NFT file" screen
- Fix the file picker logic as per the UI flow change
- Some more refactoring

### CODE REVIEW

File: lib/models/nft_format.dart [hunk 1]
```diff
@@ -5,13 +5,38 @@ import 'package:easel_flutter/utils/constants.dart';
...
   final List<String> extensions;
+  final String badge;
 
-  NftFormat({required this.format, required this.extensions});
+  NftFormat({required this.format, required this.extensions, required this.badge});
+
+  String getExtensionsList() {
+    var ret = '';
+    for (var i = 0; i < extensions.length; i++) {
+      if (i == 4) {
```
Comment (mustafapylons) [inline lib/models/nft_format.dart]: number "4" should be in a constant.

**Review by mustafapylons — COMMENTED** [review]
one small comment, otherwise code is good.

File: lib/easel_provider.dart [hunk 2]
```diff
@@ -183,7 +189,8 @@ class EaselProvider extends ChangeNotifier {
     final uploadResponse = await remoteDataSource.uploadFile(_file!);
     loading.dismiss();
     if (uploadResponse.status == Status.error) {
-      ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context).showSnackBar(SnackBar(content: Text(uploadResponse.errorMessage ?? "Upload error occurred")));
+      ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context)
+          .showSnackBar(SnackBar(content: Text(uploadResponse.errorMessage ?? "Upload error occurred")));
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Kindly move Upload error occurred in a constant file.

File: lib/easel_provider.dart [hunk 3]
```diff
@@ -251,11 +266,13 @@ class EaselProvider extends ChangeNotifier {
     log('From App $response');
 
     if (response.success) {
-      ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context).showSnackBar(const SnackBar(content: Text("Recipe created")));
+      ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context)
+          .showSnackBar(const SnackBar(content: Text("Recipe created")));
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Kindly also move Recipe created in a constant file.

File: lib/easel_provider.dart [hunk 4]
```diff
@@ -251,11 +266,13 @@ class EaselProvider extends ChangeNotifier {
...
       log("${response.data}");
       return true;
     } else {
-      ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context).showSnackBar(SnackBar(content: Text("Recipe error : ${response.error}")));
+      ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context)
+          .showSnackBar(SnackBar(content: Text("Recipe error : ${response.error}")));
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Same constant file.

File: lib/easel_provider.dart [hunk 5]
```diff
@@ -302,7 +318,8 @@ class EaselProvider extends ChangeNotifier {
       return true;
     }
 
-    ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context).showSnackBar(const SnackBar(content: Text(kStripeAccountDoesntExists)));
+    ScaffoldMessenger.of(navigatorKey.currentState!.overlay!.context)
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Can you make extension over this as it is used alot.

File: lib/screens/home_screen.dart [hunk 6]
```diff
@@ -82,52 +68,43 @@ class _HomeScreenState extends State<HomeScreen> {
...
+                            ? Consumer<EaselProvider>(
+                                builder: (_, provider, __) => TextButton(
+                                  onPressed: () {
+                                    provider.initStore();
+                                    _pageController.jumpToPage(_pageController.initialPage);
+                                  },
+                                  child: Text(
+                                    kMintMoreText,
+                                    style: Theme.of(context).textTheme.bodyText1!.copyWith(
+                                        fontSize: 20, color: EaselAppTheme.kBlue, fontWeight: FontWeight.w400),
```
Comment (kjawadDeveloper2) [inline lib/screens/home_screen.dart]: User 20.sp.

File: lib/screens/upload_screen.dart [hunk 7]
```diff
@@ -38,28 +42,62 @@ class _UploadScreenState extends State<UploadScreen> {
...
+                    child: Container(
+                      padding: EdgeInsets.symmetric(horizontal: 0.12.sw),
+                      child: PylonsButton(
+                          onPressed: () {
+                            if (provider.file != null) {
+                              context.read<EaselProvider>().priceController.clear();
+                              widget.controller
+                                  .nextPage(duration: const Duration(milliseconds: 300), curve: Curves.easeIn);
+                            } else {
+                              errorText.value = 'Pick a file';
```
Comment (kjawadDeveloper2) [inline lib/screens/upload_screen.dart]: Kindly move it to a constant file.

File: lib/screens/upload_screen.dart [hunk 8]
```diff
@@ -102,26 +140,40 @@ class _UploadWidgetState extends State<_UploadWidget> {
...
-                  widget.onFilePicked(result);
-                }
-              },
-              child: Image.asset(kFileIcon),
-            ),
+                onTap: () async {
+                  final result = await FileUtils.pickFile();
+                  if (result == null) {
+                    ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+                      content: Text("Unsupported format."),
```
Comment (kjawadDeveloper2) [inline lib/screens/upload_screen.dart]: Kindly move it to a constant file.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Kindly solve these issues.

**Review by mike-gallego — APPROVED** [review]
some constant changes, as others stated, but looks good

File: lib/easel_provider.dart (hunk 5 above)
Comment (kiroj305) [reply lib/easel_provider.dart]: @kjawadDeveloper2 , what did you mean here?

**Comment by kiroj305** [comment]
@mustafapylons , @kjawadDeveloper2 
Please review again, the issues mentioned above were fixed. Note please this is not full completion of the upload screen so I'll try to knock out all other minor issues still remained during the next PR.

**Comment by kjawadDeveloper2** [comment]
I see you have added the extension but i don't see you are using it . @Tiny-Angel305  can you confirm ?

**Comment by kiroj305** [comment]
Only used it in upload_screen which is in progress atm. As said, I am gonna fix minor issues during the next PR including the one you just pointed out. Would it be okay? @kjawadDeveloper2

**Comment by kjawadDeveloper2** [comment]
You can do snackbar fix in this and remaining in the next PR.

**Comment by kiroj305** [comment]
ok will do then.

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

**Comment by kiroj305** [comment]
@kjawadDeveloper2 , snackbar fix done.

**Review by kjawadDeveloper2 — APPROVED** [review]
Looks good from my side.

**Pull request merged**